docs/src/main/asciidoc/application-configuration-guide.adoc

```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
 MicroProfileCustomValue value;
 ----
 
+==== Converter priority
+
+In some cases you may want to use custom converter to convert type which is already converted
+by a different converter. In such case you can use `javax.annotation.Priority` annotation to
+change converters precedence and make your custom converter of higher priority then the other
+on the list.
+
+By default, if no `@Priority` can be found on a converter, it's registered with priority of 100
+and all {project-name} core converters are registered with priority of 200, so depending on which
+converter you would like to replace, you need to set higher value.
+
+To demonstrate the idea let us implement custom converter which will take precedence over 
+`MicroProfileCustomValueConverter` implemented in previous example.
+
+[source,java]
+----
+package org.acme.config;
+
+import javax.annotation.Priority;
+import org.eclipse.microprofile.config.spi.Converter;
+
+@Priority(150)
+public class MyCustomConverter implements Converter<MicroProfileCustomValue> {
+
+    @Override
+    public MicroProfileCustomValue convert(String value) {
+
+        final int secretNumber;
+        if (value.startsFrom("OBF:")) {
+            secretNumber = Integer.valueOf(SecretDecoder.decode(value));
+        } else {
+            secretNumber = Integer.valueOf(value);
+        }
+
+        return new MicroProfileCustomValue(secretNumber);
+    }
+}
+----
+
+Since it converts the same value type (namely `MicroProfileCustomValue`) and has a priority
+of 150, it will be used instead of a `MicroProfileCustomValueConverter` which has a default
+priority of 100.
+
+NOTE: This new converter also needs to be listed in a service file.
```

Comment:
Maybe you can name the service file?

Reply:
Hi @gsmet, thank you for inspecting this. All requested changes are now addressed.